### ISSUE TITLE
Minimize version duplication

### ISSUE THREAD

**Issue by barbeque-squared** [UltraStar-Deluxe/USDX#644]
There's a few different places across the sourcecode that define the version number:
- `VERSION`
- `graphics/intro-l-05.svg`
- `src/base/UConfig.pas`
- `installer/settings/variables.nsh`

In an _ideal_ world, you could just set whatever in `VERSION` and it would be reused everywhere. But it would already be a huge improvement if you could search for the contents of `VERSION`, find all literal matches, update all of them, and everything else (like whether this is a dev release etc) would just follow from that.
Note: I'm assuming any version that has a `+` anywhere is a dev version.

---

# The SVG
There is a PNG version of this SVG in `game/resources/credits/intro-l-05.png`. This would be slightly okay if that was autogenerated, but it isn't. I don't know the historical significance of the credits (there's _a lot_ of manually generated stuff in there) but **is there anything against just changing that to show `www.usdx.eu` (or something) and not have to deal with updating it ever again**?
Alternatively a PR (not by me) that the png gets autogenerated, but that's going to add build dependencies and stuff.

# UConfig.pas
This should be fairly easy to rework. I think this mostly for some places in the GUI (startup) and Error.log, and everything can be derived by either just hardcoding it near the top of the file (for easier access) or maybe even a macro to just read it at compile-time?
Note: https://wiki.freepascal.org/$include looks promising

# variables.nsh
I'd have to look up (if it's even in the git at all) what these strings should look like for a release version, but there's probably a way to make this less painful. My one question is **why is the FullVersion different from VERSION ?** `VERSION` does not have anything after the `+`, this one does. If the `dev` bit is required I'd vote to add it everywhere else as well. If not I'm fine with _either_ `+` or `+dev` (any preference?) but it should be consistent.
Note: https://stackoverflow.com/questions/15632188/how-to-load-a-files-contents-into-a-variable-define-in-nsis/15633078#15633078 looks promising
(also: what `make` command creates the Windows installer?)

---

Updating a string in three places to do a release is fine. But that should be the only work involved. Coincidentally, it would also make custom versions (for my personal builds I use build dates + git hash instead) way more trivial.

**Comment by basisbit** [comment]
USDX mostly sticks to semver 2.0.0, but strictly uses year.month.xx as release version number.

**Comment by barbeque-squared** [comment]
I've started a branch https://github.com/UltraStar-Deluxe/USDX/tree/deduplicate-versions to make this less painful to work with:
- The svg now just says `www.usdx.eu`
- `VERSION` itself should now actually be semver 2 compatible (basically, either `2020.4.0` or `2020.4.0+whatever-you-want`)
- I was unable to get `{$include}` to work (because duh, it interprets the whole thing) and I can't (or haven't yet found out how to) put these directives inside strings (the `+` in the file is an issue). But I did clean UConfig up a little bit, and at least it's now just Ctrl^F so it's still an improvement.

That leaves the installer. I know what I want to do with it (either include `VERSION`, if that's too much pain just hardcode `2020.4.0+dev` _once_. Then deduce the other stuff from it) and I've found the `makensis` command in appveyor, I'll create a PR once I've had some time to play around with it.

**Comment by barbeque-squared** [comment]
I've started a branch https://github.com/UltraStar-Deluxe/USDX/tree/deduplicate-versions to make this less painful to work with:
- The svg now just says `www.usdx.eu`
- `VERSION` itself should now actually be semver 2 compatible (basically, either `2020.4.0` or `2020.4.0+whatever-you-want`)
- I was unable to get `{$include}` to work (because duh, it interprets the whole thing) and I can't (or haven't yet found out how to) put these directives inside strings (the `+` in the file is an issue). But I did clean UConfig up a little bit, and at least it's now just Ctrl^F so it's still an improvement.

That leaves the installer. I know what I want to do with it (either include `VERSION`, if that's too much pain just hardcode `2020.4.0+dev` _once_. Then deduce the other stuff from it) and I've found the `makensis` command in appveyor, I'll create a PR once I've had some time to play around with it.
